torch/nn/modules/rnn.py

```diff
@@ -490,16 +501,14 @@ def forward(self, input, hx=None):  # noqa: F811
             max_batch_size = input.size(0) if self.batch_first else input.size(1)
             sorted_indices = None
             unsorted_indices = None
-
-        if hx is None:
-            num_directions = 2 if self.bidirectional else 1
-            hx = torch.zeros(self.num_layers * num_directions,
-                             max_batch_size, self.hidden_size,
-                             dtype=input.dtype, device=input.device)
-        else:
-            # Each batch of the hidden state should match the input sequence that
-            # the user believes he/she is passing in.
-            hx = self.permute_hidden(hx, sorted_indices)
+            if hx is None:
```

Comment:
Looks like this block of 8 lines is copied verbatim from the other branch of `if isinstance(orig_input, PackedSequence)`?
Can we move the code before the `if` and remove the duplication?

Reply:
That is not possible, unfortunately - in fact, I have moved it here from before if() . script() will fail otherwise because max_batch_size will be a Tensor in "if" conditional branch and int in 'else' (size() is traced as Tensor but treated as int when scripted - that was the reason original cast I am getting rid if, was there in the first place).

Reply:
Perhaps I should put that explanation in a comment - instead of duplicating same comment ?